Tools/ardupilotwaf/littlefs.py

```diff
@@ -0,0 +1,25 @@
+# encoding: utf-8
+
+"""
+Adds support for building littlefs as part of a Waf build
+"""
+
+from waflib.Configure import conf
+
+def configure(cfg):
+    cfg.env.append_value('GIT_SUBMODULES', 'littlefs')
+    cfg.env.prepend_value('INCLUDES', [
+        cfg.srcnode.abspath() + '/modules/littlefs/',
+    ])
+
+
+@conf
+def littlefs(bld, **kw):
+    kw.update(
+        name='littlefs',
+        source=['modules/littlefs/lfs.c', 'modules/littlefs/lfs_util.c', 'modules/littlefs/bd/lfs_filebd.c'],
+        target='littlefs',
+        defines=['LFS_NO_DEBUG'],
```

Comment:
```suggestion
        defines=['LFS_NO_DEBUG', 'LFS_NO_WARN', 'LFS_NO_ERROR', 'LFS_NO_ASSERT'],
```

Saves about 1K and disabling asserts is mandatory to avoid crashing the autopilot.

Future work: define reasonable equivalents for these functions.

Reply:
Thanks, fixed